crates/preimage/Cargo.toml

```diff
@@ -18,13 +18,12 @@ async-trait.workspace = true
 
 # local
 kona-common = { path = "../common", version = "0.0.2" }
-
-serde = { version = "1.0.203", features = ["derive"], optional = true }
+rkyv = { version = "0.7.44", optional = true }
 
 [dev-dependencies]
 tokio = { version = "1.38.0", features = ["full"] }
 tempfile = "3.10.1"
 
 [features]
 default = []
-serde = ["dep:serde"]
```

Comment:
Let's keep both `serde` and `rkyv`. Can have a mutual exclusion compile error like so in `lib.rs`.

```
#[cfg(all(feature = "serde", feature = "rkyv"))]
compile_error!("serde and rkyv are mutually exclusive feature flags and cannot be enabled together");
```